### ISSUE TITLE
closure_spacing false positive when using Swift Regex literals involving brackets

### ISSUE THREAD

**Issue by JonathanDowning** [realm/SwiftLint#4090]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

SwiftLint erroneously raises a `closure_spacing` warning when using the new Swift `Regex` literal syntax which includes an opening and closing bracket.

```swift
let r = /\{\}/ // ⚠️ Closure Spacing Violation: Closure expressions should have a single space inside each brace. (closure_spacing)
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint
main.swift:1:10: warning: Closure Spacing Violation: Closure expressions should have a single space inside each brace. (closure_spacing)

```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.47.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

```yml
only_rules
  - closure_spacing
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)? Build version 14A5294e
* Do you have a sample that shows the issue? 
```sh
echo "let r = /\{\}/" | swiftlint lint --no-cache --use-stdin --enable-all-rules
```

**Comment by jpsim** [comment]
Fixed in #4092.
